### PR TITLE
[NPU] adaptation to support operator FA3 in deterministic inference on NPU.

### PR DESCRIPTION
**Pull request by chx96642264**
## Motivation

We need to enable deterministic inference in training-inference consistency on the Ascend NPU to ensure consistent LLM outputs across runs when the --enable-deterministic-inference parameter is set, when graph mode is disabled.

## Modifications

Support the NPU operators flash_attn_npu_v3 with batch-invariant implementations to ensure deterministic inference.

The NPU operators flash_attn_npu_v3 from: [https://github.com/MinghuasLab/flash-attention-npu](url)

## Accuracy Tests

sglang startup command:
```
export ASCEND_USE_FA=1
export HCCL_DETERMINISTIC="strict"

python -m sglang.launch_server \
   --device npu \
   --attention-backend ascend \
   --sampling-backend ascend \
   --trust-remote-code \
   --tp-size 4 \
   --model-path /home/weights/Qwen3-30B-A3B \
   --disable-cuda-graph \
   --base-gpu-id 0 \
   --enable-deterministic-inference \
   --port 8879 \
   --mem-fraction-static 0.8
```

The test results are as follows:
```
python3 -m sglang.test.test_deterministic --n-trials 50 --test-mode single --port 8879
```
1-single

```
python3 -m sglang.test.test_deterministic --test-mode single --temperature 0.7 --host 127.0.0.1 --port 8879
```
2-single-temp

```
python3 -m sglang.test.test_deterministic --n-start 1 --n-trials 50 --test-mode prefix --return-logprob --port 8879
```
3-prefix

```
python3 -m sglang.test.test_deterministic --test-mode prefix --temperature 0.7 --return-logprob --host 127.0.0.1 --port 8879
```
4-prefix-temp

```
python3 -m sglang.test.test_deterministic --test-mode radix_cache --port 8879
```

5-redix_cache

```
python3 -m sglang.test.test_deterministic --test-mode radix_cache --temperature 0.7 --port 8879
```

6-redix_cache-temp

```
python3 -m sglang.test.few_shot_gsm8k --num-questions 256 --data-path /home/xx/test.jsonl --host 127.0.0.1 --port 8879
```
7-few_gsm8k

```
ais_bench --models vllm_api_stream_chat --datasets math500_gen_0_shot_cot_chat_prompt
```
9-ais_bench-math500

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #28010369668](https://github.com/sgl-project/sglang/actions/runs/28010369668)
Latest PR Test (Extra): :x: [Run #28010369461](https://github.com/sgl-project/sglang/actions/runs/28010369461)

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 import torch
 import torch_npu
+from flash_attn_npu_v3 import flash_attn_with_kvcache
```
Comment (Todobe) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py]: Does using this operator require installing additional packages? If so, it is recommended to import it when the environment variable ASCEND_USE_FA is enabled.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 2]
```diff
@@ -1203,6 +1132,51 @@ def forward_extend(
...
+                        torch.zeros(1, dtype=torch.int32).npu(),
+                        extend_seq_lens.cumsum(0).to(torch.int32),
+                    ]
+                )
+                max_seqlen_q = extend_seq_lens.max().item()
+                attn_output = flash_attn_with_kvcache(
+                    q,
+                    k,
+                    v,
+                    cache_seqlens=torch.tensor(
```
Comment (Todobe) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py]: Check if `self.forward_metadata.seq_lens` can be used.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py [hunk 3]
```diff
@@ -2162,6 +2090,33 @@ def forward_decode(
...
+                )
+                v = v_cache.view(
+                    -1, self.page_size, layer.tp_v_head_num, layer.v_head_dim
+                )
+                attn_output = flash_attn_with_kvcache(
+                    q,
+                    k,
+                    v,
+                    page_table=self.forward_metadata.block_tables,
+                    cache_seqlens=torch.tensor(
```
Comment (Todobe) [inline python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py]: ditto. Check if `self.forward_metadata.seq_lens` can be used.

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py (hunk 1 above)
Comment (chx96642264) [reply python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py]: done

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py (hunk 3 above)
Comment (chx96642264) [reply python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py]: done

File: python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py (hunk 2 above)
Comment (chx96642264) [reply python/sglang/srt/hardware_backend/npu/attention/ascend_backend.py]: done
